hotosm/fAIr
# Feature : Multimasks training for RAMP

## What does this PR do ? 

**API** 
- Adds API endpoint for training two support three new parameters input_contact_spacing , input_boundary_width and multimasks 
- Provides option in training  to choose between  binary masks and multimasks 
- Records multimasks extra param in description of training if multimasks is chosen 

**Frontend**

- Adds option for user to enable multimask feature and provide input for boundary width and contact spacing 

- Integrates https://github.com/hotosm/fAIr-utilities/pull/29 


## Screenshots 
![image](https://github.com/hotosm/fAIr/assets/36752999/c215170d-c95d-4442-9f8d-8b5169081987)


## How to test ? 

- Download preprocessed.tar.xz from training with multimasks enabled , You should have multimasks folder added 
![image](https://github.com/hotosm/fAIr/assets/36752999/96ae408a-1c60-4fe1-9552-6c760f6bf2a4)

- When you load them in to QGIS you should see multiple bands for the masks like following . You can compare with the binary masks if you like 
![image](https://github.com/hotosm/fAIr/assets/36752999/66f32d2d-4f47-40b0-8e30-0d4eca7dddd1)


- While doing training you will notice multimasks folder being used if enabled. 
- if multimask feature is disabled then things should work as it is as used to before 
- Compare the prediction with and without the boundary option ( specially in dense area )
